aws/resource_aws_kinesis_analyticsv2_application.go

```diff
@@ -0,0 +1,2177 @@
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"snapshots_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					snapshotsEnabled := strconv.FormatBool(d.Get(k).(bool))
+					return snapshotsEnabled == old
+				},
+			},
+			"flink_application_configuration": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				MaxItems:      1,
+				ConflictsWith: []string{"sql_application_configuration"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"checkpoint_configuration": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"checkpoint_interval": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"checkpointing_enabled": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"configuration_type": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateKinesisAnalyticsV2ConfigurationType,
+									},
+									"min_pause_between_checkpoints": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"monitoring_configuration": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"configuration_type": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateKinesisAnalyticsV2ConfigurationType,
+									},
+									"log_level": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateKinesisAnalyticsV2LogLevel,
+									},
+									"metrics_level": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateKinesisAnalyticsV2MetricsLevel,
+									},
+								},
+							},
+						},
+						"parallelism_configuration": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"autoscaling_enabled": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"configuration_type": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateKinesisAnalyticsV2ConfigurationType,
+									},
+									"parallelism": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"parallelism_per_kpu": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"sql_application_configuration": {
```

Comment:
Add an extra level of `application_configuration` configuration block here to match the AWS API.